services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/SwitchServiceImpl.java

```diff
@@ -77,4 +98,19 @@ void init() {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public List<Long> deleteRules(String switchId, DefaultRulesAction defaultRules, String correlationId) {
+        LOGGER.debug("Delete switch rules request received");
+
+        messageConsumer.clear();
```

Comment:
whereas we do this everywhere .. we do need to stop this practice ... prevents multiple outstanding requests to the  NB api

Reply:
Agree, but definitely shouldn't be a part of these changes. I've created a separate issue (with a potential solution) - https://github.com/telstra/open-kilda/issues/405